saleor_app_framework/app_framework/core/conf.py

```diff
@@ -1,27 +1,59 @@
 import importlib
+import logging
 import os
-
-SETTINGS_ENV_VARIABLE = "APP_SETTINGS_MODULE"
-
-
-class ImproperlyConfigured(Exception):
-    pass
-
-
-class Settings(object):
-    def __init__(self, settings_module):
-        # store the settings module in case someone later cares
-        self.SETTINGS_MODULE = settings_module
-
-        mod = importlib.import_module(self.SETTINGS_MODULE)
-
-        self._explicit_settings = set()
-        for setting in dir(mod):
-            if setting.isupper():
-                setting_value = getattr(mod, setting)
-
-                setattr(self, setting, setting_value)
-                self._explicit_settings.add(setting)
-
-
-settings = Settings(os.environ.get(SETTINGS_ENV_VARIABLE))
+from functools import lru_cache
+from typing import Optional
+
+from pydantic import BaseSettings, DirectoryPath, FilePath
+
+SETTINGS_ENV_VARIABLE = "APP_SETTINGS"
+
+
+logger = logging.getLogger(__file__)
```

Comment:
this should probably be `logger = logging.getLogger(__name__)` to maintain the parent child hierarchy of loggers